### PR TITLE
Bumped angular version from 8 to 12

### PR DESCRIPTION
**Pull request by albinscania**
**Describe pull-request**  
Upgrade from angular version 8 to 12. Some minor refactoring to comply with the new version of typescript.

**Solving issue**  
Fixes: [AB#1620](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/1620)

**How to test**  
Compare the ampliy version with the current production version and look for errors.

**Additional context**  
Added a small text file called "upgrade-notes.txt" that highlights the major steps taken. (copy pasted from CLI during update)

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -29,11 +29,18 @@ import { LandingPage } from "./components/page/landing-page/landing-page.compone
...
+    anchorScrolling: "enabled",
+    scrollOffset: [0, 144],
+    relativeLinkResolution: 'legacy'
+}; /*{
+    useHash: false,
+    scrollPositionRestoration: "enabled",
+    anchorScrolling: "enabled",
+    scrollOffset: [0, 144],
+    relativeLinkResolution: 'legacy'
+};*/
```
Comment (gaonita) [inline src/app/app-routing.module.ts:43]: duplicate comment?

File: tsconfig-es5.app.json [hunk 2]
```diff
@@ -2,5 +2,8 @@
   "extends": "./tsconfig.app.json",
   "compilerOptions": {
     "target": "es5" 
-  }
+  },
```
Comment (gaonita) [inline tsconfig-es5.app.json:5]: ^M? mysterious 😆

File: upgrade-notes.txt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  Migration completed.
+
+❯ `async` to `waitForAsync` migration.
+  The `async` testing function has been renamed to `waitForAsync` to avoid confusion with the native `async` keyword.
+  UPDATE src/app/app.component.spec.ts (1151 bytes)
+  Migration completed.
+
+❯ Removes `canActivate` from a `Route` config when `redirectTo` is also present.
+  Migration completed.
+
```
Comment (gaonita) [inline upgrade-notes.txt:50]: super! easy to follow about what's been done. thanks!

**Review by gaonita — APPROVED** [review]
Looks good to me. Awesome we finally updating! grrreat work! :)

**Review by areblk — APPROVED** [review]
No visual or behavioral regressions noticed when comparing PR to https://digitaldesign.scania.com/

**Review by theJohnnyMe — CHANGES_REQUESTED** [review]
Looks awesome and works as well. There are only two comments that Gaon added, one for extra comment and another one that looks like a typo. Please check them and ping us after :)

**Comment by albinscania** [comment]
Haha yeah, some auto merger-weirdness snuck in! Fixed now :)
Good find.

**Review by theJohnnyMe — APPROVED** [review]
Great! Thanks a lot Albin for sorting out this one. Go and "Squash and merge" it :)

**Pull request merged**